test/uk/gov/hmrc/universalcreditliabilitystubs/controllers/UcLiabilityControllerSpec.scala

```diff
@@ -16,27 +16,46 @@
 
 package uk.gov.hmrc.universalcreditliabilitystubs.controllers
 
+import org.mockito.ArgumentMatchers
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.when
 import org.scalatest.matchers.must.Matchers
 import org.scalatest.wordspec.AnyWordSpec
+import org.scalatestplus.mockito.MockitoSugar.mock
 import play.api.http.Status
+import play.api.test.Helpers
 import play.api.test.Helpers.*
-import play.api.test.{FakeRequest, Helpers}
+import uk.gov.hmrc.universalcreditliabilitystubs.services.UcLiabilityService
+import uk.gov.hmrc.universalcreditliabilitystubs.support.TestHelpers
 
-class UcLiabilityControllerSpec extends AnyWordSpec with Matchers {
+class UcLiabilityControllerSpec extends AnyWordSpec with TestHelpers with Matchers {
 
-  private val fakeRequest = FakeRequest("POST", "/")
-  private val controller  = new UcLiabilityController(Helpers.stubControllerComponents())
+  private val mockUcLiabilityService = mock[UcLiabilityService]
+  private val controller             = new UcLiabilityController(Helpers.stubControllerComponents(), mockUcLiabilityService)
 
   "insertLiabilityDetails" must {
     "return 204" in {
-      val result = controller.insertLiabilityDetails("nino")(fakeRequest)
+
+      val fakeRequest = generateFakeRequest(validInsertLiabilityRequest, validHeaders)
+
+      when(mockUcLiabilityService.validateInsertLiabilityRequest(any(), any()))
+        .thenReturn(Right(getInsertLiabilityRequest))
+
+      val result =
+        controller.insertLiabilityDetails(generateNino())(fakeRequest)
       status(result) mustBe Status.NO_CONTENT
     }
   }
```

Comment:
Not too clear what the long term plan for this is.
We need either a lot more tests here, or to remove it altogether

Not sure what additional value it would have currently (since IT covers everything here and more)